include/eve/detail/function/swap_adjacent_groups.hpp

```diff
@@ -18,32 +18,33 @@ namespace eve
 {
   EVE_MAKE_CALLABLE(swap_adjacent_groups_, swap_adjacent_groups);
 
-  //================================================================================================
-  // Premade pattern generator
-  template<std::ptrdiff_t G, std::ptrdiff_t N>
-  inline constexpr
-  auto swap_adjacent_groups_n = fix_pattern<N> ( [](auto i, auto)
-                                              {
-                                                if(G!=N)  return (i+G)%(G*2) + (G*2)*(i/(G*2));
-                                                else      return i;
-                                              }
-                                            );
-
-  //================================================================================================
-  // Classify a pattern as a swag
-  template<std::ptrdiff_t... I> inline constexpr auto is_swag = []()
+  namespace detail
   {
-    // List all possible swags for a current size
-    constexpr auto sz = sizeof...(I);
-    constexpr auto x = []<std::size_t... N>( std::index_sequence<N...> )
+    //==============================================================================================
+    // Classify a pattern as a swag
+    template<std::ptrdiff_t G, std::ptrdiff_t N>
+    inline constexpr
+    auto swap_adjacent_groups_n = fix_pattern<N> ( [](auto i, auto)
```

Comment:
Can we call these `*_pattern`? `_n` doesn't tell me anything.

Reply:
sure